Settings initial commit

contrlbutor opened the pull request:
Начальный коммит страницы настроек. Пока что без правок стилей antd и не очень по сетке изначального дизайна.

Commit:
Settings initial commit

Commit:
code style fixes

Commit:
Merge branch 'develop' into settings

Commit:
settings style fixes (partial)

Commit:
style fixes (partial)

Commit:
style fixes (partial)

shiryaev585 commented on frontend/src/pages/Settings/Settings.jsx:
```diff
@@ -0,0 +1,76 @@
+                <Title className="settings__header--title" level={2}>Настройки</Title>
+            </div>
+            <div className="settings__content">
+                <Button type="secondary" size="large" className="settings__content--login">
```
Есть смысл сделать здесь не кнопкой, а ссылкой на страницу входа

awesomesk1ll replied:
Ага, ну это пока чисто визуальный набросок, а так да. Всё верно.

vladsmit commented on frontend/src/pages/Settings/Settings.jsx:
```diff
@@ -0,0 +1,76 @@
+
+const ratings = { 5: 'от 5', 10: 'до 10' };
+const years = { 1980: 'c 1980', 2021: 'по 2021' };
+const genres = [{ value: 'боевик' }, { value: 'комедия' }, { value: 'драма' }, { value: 'мюзикл' }];
```
- В настройках нужно увеличить количество выборки по жанрам и странам
- При скролле карточки, иногда, вместо прокрутки описания цепляется постер и оттягивается

awesomesk1ll replied:
Да, я все доступные варианты добавлю, как доделывать сяду.

awesomesk1ll replied:
про пункт 2 не очень понял @vladsmit поясни

awesomesk1ll replied:
Это к данному пулл реквесту разве относится ? :)

Commit:
big style changes (dark theme, antd patches)

1. renamed ThemeButton component to ThemeSwitch
2. slighty fixed ThemeSwitch component
3. moved "dark theme" and "antd patches" parts from style.scss to separate files.

Commit:
filled options, created handlers

1. expanded genres and countries as actual
2. added handlers to pick/unpick all data available
3. write in select disabled
so now mobile keyboard not opens on dropdown
4. moved form data bank to config.js file
